pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,832 @@
+    mostly in the first quadrant, and interval analysis guarantees the
+    numerical computations up to a specified tolerance.
+
+    New gold data can be added to the returned dict. Each change should be
+    versioned in ``dataset_version``, following SemVer and maintaining
+    backwards compatibility whenever possible. Be sure to update all dependent
+    functions.
+
+    Parameters
+    ----------
+
+    Returns
+    -------
+    gold_dataset : dict
+
+    References
+    ----------
+    [1] J. W. Bishop, "Computer simulation of the effects of electrical
+    mismatches in photovoltaic cell interconnection circuits", Solar Cells,
+    vol 25, pp. 73-89, 1988.
+    """
+
+    # Load module inside function to avoid a global dependency requirement
+    # http://pyinterval.readthedocs.io/en/latest/guide.html
+    # Use `pip install pyinterval`, even in conda distro's
+    from interval import interval, imath
+
+    # Approximate largest possible absolute residual value, determined
+    #  empirically
+    tol = 1.e-12
+
+    tol_interval = interval([-tol, tol])
+
+    # Generate I-V curves, each with same number of points, over matrix of
+    #  (G,T) combinations
+    G_list = [100., 200., 400., 600., 800., 1000., 1100.]  # W/m^2
+    T_list = [15., 25., 50., 75.]  # degC
+    NUM_PTS = 21
+
+    # The "base" devices, having ideal, normal, and degraded versions
+    devices = [
+        bootstrap_si_device_bishop(),
+        bootstrap_si_device_bishop(N_s=8),
+        bootstrap_si_device_pvmismatch(),
+        pvsystem.retrieve_sam('cecmod').SunPower_SPR_E20_327,
+        pvsystem.retrieve_sam('cecmod').First_Solar_FS_495
+    ]
+
+    # Initialize the output, recording metadata
+    gold_dataset = {
+        "timestamp_utc_iso": datetime.datetime.utcnow().isoformat(),
+        "dataset_version": '1.0.0',  # dataset follows SemVer
+        "python_version": sys.version,
+        "numpy_version": numpy.__version__,
+        "pvlib_version": pvlib.version.__version__,
+        "platform_version": platform.platform(),
+        "residual_interval_tolerance": tol,
+        "units": {
+            "G": 'W/m^2',
+            "T": 'degC',
+            "i_l": 'A',
+            "i_0": 'A',
+            "r_s": 'Ohm',
+            "g_sh": 'S',
+            "nNsVth": 'V',
+            "v_gold": 'V',
+            "i_gold": 'A'
+        },
+        "devices": []
+    }
+
+    # For each ideal, nominal, or degraded device, create matrix of I-V curves
+    for device in devices:
+        device_dict = {
+            "name": device.name,
+            "Technology": device.Technology,
+            "N_s": device.N_s,
+            "iv_curves": []
+        }
+
+        # Choose bandgap parameters based on device technology
+        # Add new technologies if/when needed, should match a CEC module
+        #  database type
+        # These are NOT saved, but are versioned with the dataset version
+        if device.Technology == 'Mono-c-Si' or \
```

Comment:
Suggest moving this ``if`` block to the ``create_device_dict`` or similar function requested in the loop below. This block could also be a small function:

```python
def _determine_bandgap_parameters(device):
    if...elif...else...
    return bandgap_params  # a dict

# within "create_device_dict"
pvsystem.calcparams_desoto(..., **bandgap_params)
```